lib/inc/drogon/HttpAppFramework.h

```diff
@@ -171,15 +171,18 @@ class DROGON_EXPORT HttpAppFramework : public trantor::NonCopyable
      * be sent to the client to provide a custom layout.
      */
     virtual HttpAppFramework &setCustomErrorHandler(
-        std::function<HttpResponsePtr(HttpStatusCode)> &&resp_generator) = 0;
+        std::function<HttpResponsePtr(HttpStatusCode,
+                                      const HttpRequestPtr &req)>
```

Comment:
I prefer to add a new method instead of changing the old one. We should keep the public API stable for old version users.

```c++
virtual HttpAppFramework &setCustomErrorHandler(
        std::function<HttpResponsePtr(HttpStatusCode,
                                      const HttpRequestPtr &req)> &&resp_generator) = 0;
HttpAppFramework &setCustomErrorHandler(
        std::function<HttpResponsePtr(HttpStatusCode)> &&resp_generator)
{
    return setCustomErrorHandler([cb=std::move(resp_generator)] 
                                      (HttpStatusCode code,
                                      const HttpRequestPtr &req){
                                                   return cb(code);
                                       });
}
```